### PR TITLE
br: wait old backup job done before start

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

fix backup error with auto retry
close https://github.com/pingcap/tidb-operator/issues/4980

### What is changed and how does it work?

* not start new backup job until the old job was deleted
* refine the log of backup retry part

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
fix an bug that auto backup retry failed
```

### CODE REVIEW

File: pkg/backup/backup/backup_manager.go [hunk 1]
```diff
@@ -189,34 +189,28 @@ func (bm *backupManager) skipBackupSync(backup *v1alpha1.Backup) (bool, error) {
 	return bm.skipSnapshotBackupSync(backup)
 }
 
-// waitPreTaskDone waits pre task done.
-// just log backup needs to wait pre task done. Mainly including:
+// waitPreTaskDone waits pre task done. this can make sure there is only one backup job running.
 // 1. wait other command done, such as truncate/stop wait start done.
 // 2. wait command's job done
 func (bm *backupManager) waitPreTaskDone(backup *v1alpha1.Backup) error {
```
Comment (Ehco1996) [inline pkg/backup/backup/backup_manager.go:195]: this change can fix the issue

**Comment by Ehco1996** [comment]
the e2e test fiailed

```

[2023-04-24T04:10:46.997Z] [3] • Failure [103.095 seconds]

[2023-04-24T04:10:46.997Z] [3] Backup and Restore

[2023-04-24T04:10:46.997Z] [3] /home/jenkins/agent/workspace/tidb-operator-pull-e2e-kind-br/go/src/github.com/pingcap/tidb-operator/tests/e2e/br/br.go:122

[2023-04-24T04:10:46.997Z] [3]   Restart Backup by k8s Test

[2023-04-24T04:10:46.997Z] [3]   /home/jenkins/agent/workspace/tidb-operator-pull-e2e-kind-br/go/src/github.com/pingcap/tidb-operator/tests/e2e/br/br.go:507

[2023-04-24T04:10:46.997Z] [3]     delete backup pod and restart by k8s test [It]

[2023-04-24T04:10:46.997Z] [3]     /home/jenkins/agent/workspace/tidb-operator-pull-e2e-kind-br/go/src/github.com/pingcap/tidb-operator/tests/e2e/br/br.go:508

[2023-04-24T04:10:46.997Z] [3] 

[2023-04-24T04:10:46.997Z] [3]     Unexpected error:

[2023-04-24T04:10:46.997Z] [3]         <*errors.errorString | 0xc0002d6c10>: {

[2023-04-24T04:10:46.997Z] [3]             s: "can't wait for backup complete: backup is failed, reason: BackupDataToRemoteFailed, message: cluster br-3827/delete-backup-pod-test, wait pipe message failed, errMsg , err: signal: killed",

[2023-04-24T04:10:46.997Z] [3]         }

[2023-04-24T04:10:46.997Z] [3]         can't wait for backup complete: backup is failed, reason: BackupDataToRemoteFailed, message: cluster br-3827/delete-backup-pod-test, wait pipe message failed, errMsg , err: signal: killed

[2023-04-24T04:10:46.997Z] [3]     occurred

[2023-04-24T04:10:46.997Z] [3] 

[2023-04-24T04:10:46.997Z] [3]     /home/jenkins/agent/workspace/tidb-operator-pull-e2e-kind-br/go/src/github.com/pingcap/tidb-operator/tests/e2e/br/br.go:544

[2023-04-24T04:10:46.997Z] [3] 

[2023-04-24T04:10:46.997Z] [3]     Full Stack Trace

[2023-04-24T04:10:46.997Z] [3]     github.com/pingcap/tidb-operator/tests/e2e/br.glob..func1.10.1()

[2023-04-24T04:10:46.997Z] [3]     	/home/jenkins/agent/workspace/tidb-operator-pull-e2e-kind-br/go/src/github.com/pingcap/tidb-operator/tests/e2e/br/br.go:544 +0x2af

[2023-04-24T04:10:46.997Z] [3]     github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc00143f340?)

[2023-04-24T04:10:46.997Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/internal/leafnodes/runner.go:113 +0xb1

[2023-04-24T04:10:46.997Z] [3]     github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc00143f5f0?)

[2023-04-24T04:10:46.997Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/internal/leafnodes/runner.go:64 +0x125

[2023-04-24T04:10:46.997Z] [3]     github.com/onsi/ginkgo/internal/leafnodes.(*ItNode).Run(0x7d?)

[2023-04-24T04:10:46.997Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/internal/leafnodes/it_node.go:26 +0x7b

[2023-04-24T04:10:46.997Z] [3]     github.com/onsi/ginkgo/internal/spec.(*Spec).runSample(0xc0015d1950, 0xc00143f9b8?, {0x3b9e640, 0xc000182e80})

[2023-04-24T04:10:46.997Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/internal/spec/spec.go:215 +0x2a9

[2023-04-24T04:10:46.997Z] [3]     github.com/onsi/ginkgo/internal/spec.(*Spec).Run(0xc0015d1950, {0x3b9e640, 0xc000182e80})

[2023-04-24T04:10:46.997Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/internal/spec/spec.go:138 +0xf2

[2023-04-24T04:10:46.997Z] [3]     github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpec(0xc0015a37c0, 0xc0015d1950)

[2023-04-24T04:10:46.997Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/internal/specrunner/spec_runner.go:200 +0xf1

[2023-04-24T04:10:46.998Z] [3]     github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpecs(0xc0015a37c0)

[2023-04-24T04:10:46.998Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/internal/specrunner/spec_runner.go:170 +0x1b6

[2023-04-24T04:10:46.998Z] [3]     github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc0015a37c0)

[2023-04-24T04:10:46.998Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/internal/specrunner/spec_runner.go:66 +0xc5

[2023-04-24T04:10:46.998Z] [3]     github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc0001c5ce0, {0x7f83c39d9518, 0xc000703520}, {0x35af0b1, 0x17}, {0xc0010135e0, 0x2, 0x2}, {0x3bc8d40, 0xc000182e80}, ...)

[2023-04-24T04:10:46.998Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/internal/suite/suite.go:79 +0x4e5

[2023-04-24T04:10:46.998Z] [3]     github.com/onsi/ginkgo.RunSpecsWithCustomReporters({0x3ba08c0?, 0xc000703520}, {0x35af0b1, 0x17}, {0xc0010135c0, 0x2, 0x35dd6a6?})

[2023-04-24T04:10:46.998Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/ginkgo_dsl.go:219 +0x189

[2023-04-24T04:10:46.998Z] [3]     github.com/onsi/ginkgo.RunSpecsWithDefaultAndCustomReporters({0x3ba08c0, 0xc000703520}, {0x35af0b1, 0x17}, {0xc001368820, 0x1, 0x1})

[2023-04-24T04:10:46.998Z] [3]     	/go/pkg/mod/github.com/onsi/ginkgo@v1.14.1/ginkgo_dsl.go:207 +0x10a

[2023-04-24T04:10:46.998Z] [3]     github.com/pingcap/tidb-operator/tests/e2e.RunE2ETests(0x48eb37?)

[2023-04-24T04:10:46.998Z] [3]     	/home/jenkins/agent/workspace/tidb-operator-pull-e2e-kind-br/go/src/github.com/pingcap/tidb-operator/tests/e2e/e2e.go:489 +0x3b9

[2023-04-24T04:10:46.998Z] [3]     github.com/pingcap/tidb-operator/tests/e2e.TestE2E(0x406ab9?)

[2023-04-24T04:10:46.998Z] [3]     	/home/jenkins/agent/workspace/tidb-operator-pull-e2e-kind-br/go/src/github.com/pingcap/tidb-operator/tests/e2e/e2e_test.go:139 +0x19

[2023-04-24T04:10:46.998Z] [3]     testing.tRunner(0xc000703520, 0x37070f0)

[2023-04-24T04:10:46.998Z] [3]     	/usr/local/go/src/testing/testing.go:1446 +0x10b

[2023-04-24T04:10:46.998Z] [3]     created by testing.(*T).Run

[2023-04-24T04:10:46.998Z] [3]     	/usr/local/go/src/testing/testing.go:1493 +0x35f

[2023-04-24T04:10:46.998Z] [3] ------------------------------

[2023-04-24T04:10:46.998Z] [3] Apr 24 04:10:46.708: INFO: Running AfterSuite actions on all nodes
```

**Comment by Ehco1996** [comment]
seems unteable test https://ci.pingcap.net/blue/organizations/jenkins/tidb-operator-pull-e2e-kind-br/detail/tidb-operator-pull-e2e-kind-br/2677/pipeline/

File: pkg/controller/backup/backup_controller.go [hunk 2]
```diff
@@ -395,42 +396,48 @@ func (c *Controller) isBackupPodOrJobFailed(backup *v1alpha1.Backup) (bool, stri
 	selector, err := label.NewBackup().Instance(backup.GetInstanceName()).BackupJob().Backup(name).Selector()
 	if err != nil {
 		klog.Errorf("Fail to generate selector for backup %s/%s, error %v", ns, name, err)
-		return false, "", "", err
```
Comment (grovecai) [inline pkg/controller/backup/backup_controller.go:398]: actually, I prefer old style return, it's more directly

File: pkg/controller/backup/backup_controller.go [hunk 3]
```diff
@@ -259,41 +259,41 @@ func (c *Controller) enqueueBackup(obj interface{}) {
...
 	}
 
 	// check whether backup pod and job failed
-	isPodOrJobFailed, reason, originalReason, err = c.isBackupPodOrJobFailed(backup)
+	podOrJobFailed, reason, originalReason, err = c.isBackupPodOrJobFailed(backup)
 	if err != nil {
 		klog.Errorf("Fail to check backup %s/%s pod or job status, %v", ns, name, err)
-		return false, "", "", err
+		podOrJobFailed = false
+		return
```
Comment (grovecai) [inline pkg/controller/backup/backup_controller.go:282]: I prefer old style return, it's more directly

**Review by grovecai — APPROVED** [review]

**Comment by grovecai** [comment]
rest lgtm

File: pkg/controller/backup/backup_controller.go (hunk 3 above)
Comment (Ehco1996) [reply pkg/controller/backup/backup_controller.go:282]: IMO, it's actually the opposite. For example, the previous return statement was like this: `false, "", "", err`.

who knows what is `""` or false?

**Review by grovecai — APPROVED** [review]

**Pull request merged**